### ISSUE TITLE
Unexpected status code 401: Unauthorized, Could not parse JWT

### ISSUE THREAD

**Issue by ajtsch** [pusher/docs#82]
Hello, 

I am pretty sure I am on the wrong here but we never know. I think I spotted an issue on https://pusher.com/docs/beams/guides/publish-to-specific-user/web/.

My JS code (as on the doc)

`
const beamsClient = new PusherPushNotifications.Client({
		instanceId: 'MY_KEY',
	  });
	  const beamsTokenProvider = new PusherPushNotifications.TokenProvider({
			url: "MY_API_URL"
		  });
	  beamsClient
		.start()
		.then(() => beamsClient.setUserId("user_1", beamsTokenProvider))
		.catch(console.error);
`

returns the following console errors : 

`push-notifications-cdn.js:889 PUT https://3759f479-0cba-4c4c-8c2e-0653dbe43953.pushnotifications.pusher.com/device_api/v1/instances/3759f479-0cba-4c4c-8c2e-0653dbe43953/devices/web/web-9e20dc2a-da22-46ea-89c4-abe51b989828/user 401 (Unauthorized)`

And

`Error: Unexpected status code 401: Unauthorized, Could not parse JWT
    at _callee2$ (push-notifications-cdn.js:968)
    at tryCatch (push-notifications-cdn.js:82)
    at Generator.invoke [as _invoke] (push-notifications-cdn.js:308)
    at Generator.prototype.<computed> [as next] (push-notifications-cdn.js:134)
    at asyncGeneratorStep (push-notifications-cdn.js:770)
    at _next (push-notifications-cdn.js:792)`

Any clue ? Thank you very much. If you need additional info please let me know

**Comment by benw-pusher** [comment]
Hi, 

Could you share the token generated by your token provider? The error would indicate the token is malformed or incorrect in some way.

**Comment by ajtsch** [comment]
Sure, my api returns the following json :

`"eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwczpcL1wvMzc1OWY0NzktMGNiYS00YzRjLThjMmUtMDY1M2RiZTQzOTUzLnB1c2hub3RpZmljYXRpb25zLnB1c2hlci5jb20iLCJzdWIiOiJ1c2VyLTEiLCJleHAiOjE2Mjc2MzI2Njl9.cOdb7XuEw5cFG9ICfzYkacONoYNuh7SUHh5S-uzvcr4"` 

Thank you !

**Comment by benw-pusher** [comment]
Just to be clear, is the 401 error returned all the time or only when the response is 
```
{
  "message": "No token has been found"
}
```

**Comment by ajtsch** [comment]
No, the code **401 on PUT** is returned only when the token (sent by my API) is **correct**.

If my API sends back the wrong token (or the error message you cited), the console displays the http error 400 on GET. Which makes sense.

Summary :
- Wrong token sent back by my API => "HTTP 400 on GET"
- Correct token sent back by my API => "HTTP 401 on PUT"

Is it clearer ?

Thank you

**Comment by benw-pusher** [comment]
Thanks for clarifying. Are you able to share the code that generates the authToken? If you would prefer to share privately then feel free to reach out via support at https://support.pusher.com/hc/en-us/requests/new.

**Comment by ajtsch** [comment]
Thank you, I'll contact the support with the link above :)
